### ISSUE TITLE
Support for partitioned tables 

### ISSUE THREAD

**Issue by paurora17** [Mooncake-Labs/pg_mooncake#17]
### What feature are you requesting?

Today, `CREATE TABLE ... USING columnstore` is incompatible with partitioned tables.

### Why are you requesting this feature?

Want to keep a history of up to 90 days in the DB, but to start archiving data after that period.

**Comment by pantonis** [comment]
It would be good to have this. One of the 2 things that keep us from using pg_mooncake

**Comment by zhousun** [comment]
https://github.com/Mooncake-Labs/pg_mooncake/commit/373bfc5d6cd8f425b7bab069e2e249e8aecc092c
should cover most use cases of partitioned tables in postgres.

For time series use cases, each file will likely only contain data from similar time range.
So at query time, based on the filter, most data files will be skipped.

**Comment by ragnorc** [comment]
What about splitting by custom column? E.g. tenant id

**Comment by mike-luabase** [comment]
do you need pushdown for iceberg in duckdb before this will work?

**Comment by zhousun** [comment]
> What about splitting by custom column? E.g. tenant id
yes, this is a very valid use case. But instead of partitioned table, we will likely implement a sort key instead (to avoid exaggerating tiny file problem) 

> do you need pushdown for iceberg in duckdb before this will work?
we implemented our customized scan which does predicate push down & skip.

**Comment by dpxcc** [comment]
v0.2 now supports partitioned tables: https://github.com/Mooncake-Labs/pg_mooncake/commit/a1eb01c7075e2d5e4896bf73febc46f85fda2ad2
